src/series/mod.rs

```diff
@@ -44,7 +44,7 @@ impl Series {
     }
     pub fn as_physical(&self) -> DaftResult<Series> {
         let physical_dtype = self.data_type().to_physical();
-        if &physical_dtype == self.data_type() {
+        if &physical_dtype == self.data_type() || self.data_type().is_extension() {
```

Comment:
Why have this fallthrough?

Reply:
Ah this was previously required when `DataType::Extension(_, inner, _).to_physical()` returned `inner.to_physical()` but that's no longer the case with the new logical types refactor, so this fallthrough should actually be removed.

Good catch! 🙌